### PR TITLE
Ability to map a signal component instead of picking from list

### PR DESCRIPTION
**Pull request by johnclary**
This PR depends on https://github.com/cityofaustin/atd-moped/pull/1094.

## Associated issues
- https://github.com/cityofaustin/atd-data-tech/issues/12998

## Testing
**URL to test:**  Local

1. Create a signal by picking it from the signal autocomplete input (should work normally)
2. Edit the signal component by clearing the selected signal from the autocomplete input (should result in unmapped component)
3. Map the unmapped signal component (location should be saved as expected)
4. Edit the the signal component you just mapped by picking a signal from the list (the location should be updated with the location you selected)
5. Edit the signal component again by changing it to an **Intersection - Improvement** component (this will result in the component being unmapped)
6. Edit the **Intersection - Improvement** component by switching to a signal component, but leave the signal input blank (should result in the location being unchanged).

---
#### Ship list
- [ ] Code reviewed 
- [ ] Product manager approved
- [ ] Added to [QA test script](https://docs.google.com/spreadsheets/d/1n_O6MLh9cwwPf57HUM394Ea-z9uuoEV1-QW4axNZXLE/edit#) if applicable

### CODE REVIEW

File: moped-database/migrations/1696693713827_asset_feature_layer_id/up.sql
Comment (johnclary) [inline moped-database/migrations/1696693713827_asset_feature_layer_id/up.sql:15]: Signal and phb components will reference the  `feature_intersections` layer just like all other point components, and now there is a new reference column (`asset_feature_layer_id`) specifically for which layer to use for asset data.

This should make easier to scale out to additional asset types like flashing beacons and school zones (both of which have been requested by AMD).

File: moped-editor/src/views/projects/projectView/ProjectComponents/ComponentForm.js
Comment (johnclary) [inline moped-editor/src/views/projects/projectView/ProjectComponents/ComponentForm.js:123]: I don't think it's big deal, but we still have some awkward hardcoding of component types and layer names in the editor. Maybe we can try to abstract this a bit when we look to add additional asset types (e.g. flashing beacons).

**Comment by johnclary** [comment]
👋 folks, this is ready for review, but the diff is going to be impossible until I merge https://github.com/cityofaustin/atd-moped/pull/1094. 

Feel free to hold off on code review 🤷

File: moped-editor/src/views/projects/projectView/ProjectComponents/utils/makeComponentData.js
Comment (mddilley) [inline moped-editor/src/views/projects/projectView/ProjectComponents/utils/makeComponentData.js:128]: This all looks solid, but I found a small bug where a previously deleted signal features remains in state. To reproduce:
1. Create a signal component with a chosen signal from the signal dropdown
2. Click on the map so that no list items are expanded
3. Click on your signal component feature on the map to expand its associated list item
4. Edit attributes and clear the chosen signal from the form. Then, save
5. Click the red map edit button and you will see the previously deleted signal feature in yellow on the map

https://github.com/cityofaustin/atd-moped/assets/37249039/ad620ab4-8f35-4ff4-a13e-b5acdfdd60b0

**Review by mddilley — CHANGES_REQUESTED** [review]
Went ahead and reviewed this since the other PR changes were fresh in my mind to skip over. All works great, and this is a huge code improvement! 🚀 I can easily switch between signal features from the socrata dataset, drawn or selected features, and no features (unmapped). I found a small bug - let me know if you want to bounce around some ideas about what is causing it.

File: moped-editor/src/views/projects/projectView/ProjectComponents/EditAttributesModal.js
Comment (johnclary) [reply moped-editor/src/views/projects/projectView/ProjectComponents/EditAttributesModal.js:96]: thanks—i forgot i made this note 😵 

looking at the signal autocomplete, it looks like we don't actually register the form with RFH [until the options are loaded](https://github.com/cityofaustin/atd-moped/blob/458ed9e98250e16f7374cee6cf31b2408bfd9f85/moped-editor/src/views/projects/projectView/ProjectComponents/SignalComponentAutocomplete.js#L45) - so i think it's actually pretty harmless if the socrata request fails. the user cannot pick a signal component, but, if they are editing, the previously saved signal component is still persisted in state.

File: moped-editor/src/views/projects/projectView/ProjectComponents/utils/makeComponentData.js
Comment (mddilley) [reply moped-editor/src/views/projects/projectView/ProjectComponents/utils/makeComponentData.js:128]: just tested this out, and it looks great. Having all of that logic in one place is the way! 🙌

**Review by mddilley — APPROVED** [review]
The updated syncing of the clicked and draft edit states looks great. Glad to see those reined in! 🚢

File: moped-editor/src/queries/components.js
Comment (chiaberry) [inline moped-editor/src/queries/components.js:205]: its called update component attributes everywhere else but here

**Review by chiaberry — APPROVED** [review]
I think this will make AMD happy!

**Pull request merged**
